### PR TITLE
[Test][sample-yaml] Check RayCluster create correct number of pods

### PR DESCRIPTION
**Pull request by MortalHappiness**
## Why are these changes needed?

See the description in the corresponding issue for details.

Note:

- Tests are changed into whitelist manner, only those sample yaml files that need to be checked are listed.
- Currently I find it hard to decode yaml files containing multiple resources, so I moved `RayCluster` resource to the top if there are multiple resources declared in a yaml file. 

## Related issue number

Closes: ray-project/kuberay#2431

## Checks

- [x] I've made sure the tests are passing.
- Testing Strategy
   - [x] Unit tests
   - [ ] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by MortalHappiness** [comment]
@kevin85421 @andrewsykim PTAL

File: ray-operator/test/sampleyaml/raycluster_test.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+			rayCluster = GetRayCluster(test, namespace.Name, rayCluster.Name)
+
+			// Check if the RayCluster created correct number of pods
+			var desiredWorkerReplicas int32
+			if rayCluster.Spec.WorkerGroupSpecs != nil {
+				for _, workerGroupSpec := range rayCluster.Spec.WorkerGroupSpecs {
+					desiredWorkerReplicas += *workerGroupSpec.Replicas
+				}
+			}
+			test.Expect(rayCluster.Status.DesiredWorkerReplicas).To(gomega.Equal(desiredWorkerReplicas))
```
Comment (kevin85421) [inline ray-operator/test/sampleyaml/raycluster_test.go:83]: Can you check if the head Pod is running and ready, and if all worker Pods are running and ready as well?

Comment (MortalHappiness) [reply ray-operator/test/sampleyaml/raycluster_test.go:83]: done

**Comment by kevin85421** [comment]
@MortalHappiness can you fix the CI errors?

File: ray-operator/test/sampleyaml/raycluster_test.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+			// Check if the head pod is ready
```
Comment (kevin85421) [inline ray-operator/test/sampleyaml/raycluster_test.go:85]: Can you verify the number of head Pods and worker Pods, which are the source of truth?

**Review by kevin85421 — APPROVED** [review]

File: ray-operator/test/sampleyaml/raycluster_test.go (hunk 2 above)
Comment (MortalHappiness) [reply ray-operator/test/sampleyaml/raycluster_test.go:86]: done

**Review by andrewsykim — APPROVED** [review]

**Comment by andrewsykim** [comment]
Looks like ` Test RayCluster Sample YAMLs (nightly operator)` is still failing, is it supposed to be passing after this PR
```

[2024-10-17T02:23:20Z] ======================================================================
--
  | [2024-10-17T02:23:20Z] ERROR: runtest (framework.prototype.GeneralTestCase.runtest)
  | [2024-10-17T02:23:20Z] Run a configuration test
  | [2024-10-17T02:23:20Z] ----------------------------------------------------------------------
  | [2024-10-17T02:23:20Z] Traceback (most recent call last):
  | [2024-10-17T02:23:20Z]   File "/workdir/tests/framework/prototype.py", line 586, in runtest
  | [2024-10-17T02:23:20Z]     self.cr_event.trigger()
  | [2024-10-17T02:23:20Z]   File "/workdir/tests/framework/prototype.py", line 188, in trigger
  | [2024-10-17T02:23:20Z]     self.wait()
  | [2024-10-17T02:23:20Z]   File "/workdir/tests/framework/prototype.py", line 399, in wait
  | [2024-10-17T02:23:20Z]     raise Exception("RayClusterAddCREvent wait() timeout")
  | [2024-10-17T02:23:20Z] Exception: RayClusterAddCREvent wait() timeout
  | [2024-10-17T02:23:20Z]
  | [2024-10-17T02:23:20Z] ----------------------------------------------------------------------
  | [2024-10-17T02:23:20Z] Ran 17 tests in 1141.390s
  | [2024-10-17T02:23:20Z]
  | [2024-10-17T02:23:20Z] FAILED (errors=1)
  | [2024-10-17T02:23:20Z] Traceback (most recent call last):
  | [2024-10-17T02:23:20Z]   File "/workdir/tests/test_sample_raycluster_yamls.py", line 91, in <module>
  | [2024-10-17T02:23:20Z]     assert test_result.wasSuccessful()
  | [2024-10-17T02:23:20Z] AssertionError
  | [2024-10-17T02:23:20Z] 🚨 Error: The command exited with status 1


```

**Comment by MortalHappiness** [comment]
@andrewsykim The failed test is python test, which is deprecated. Those deprecated tests will be removed once all python tests are rewritten with golang.

**Pull request merged**